core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -1458,6 +1458,17 @@ protected Object visitFormat(Format node, Object context)
                     .map(this::type)
                     .collect(toImmutableList());
 
+            List<Object> processedArguments = arguments.stream()
+                    .map(argument -> processWithExceptionHandling(argument, context))
+                    .collect(toList());
+
+            if (hasUnresolvedValue(processedArguments)) {
+                return new Format(ImmutableList.<Expression>builder()
```

Comment:
It's ok. There's no benefit in returning the desugared expressions since they will get translated during planning. In fact, it means we're duplicating behavior in two places.